src/type_checker/checker.rs

```diff
@@ -727,7 +727,7 @@ impl<B: Backend> TypeChecker<B> {
                 ));
             }
 
-            if !typ.match_expected(&sig_arg.typ.kind) {
+            if !typ.match_expected(&sig_arg.typ.kind, true) {
```

Comment:
the arguments can be either constant or not, unless the signature specifies the const attribute.